source/slang/slang-ast-val.cpp

```diff
@@ -2115,4 +2116,79 @@ Val* IntVal::linkTimeResolve(Dictionary<String, IntVal*>& mapMangledNameToVal)
     SLANG_AST_NODE_VIRTUAL_CALL(IntVal, linkTimeResolve, (mapMangledNameToVal));
 }
 
+//
+// UIntSetVal
+//
+
+UIntSet UIntSetVal::toUIntSet() const
+{
+    UIntSet result;
+
+    // Convert each bitmask operand back to UIntSet::Element
+    for (Index i = 0; i < getBitmaskCount(); i++)
+    {
+        auto bitmask = getBitmask(i);
+        if (bitmask)
+        {
+            auto element = static_cast<UIntSet::Element>(bitmask->getValue());
+            result.addRawElement(element, i);
+        }
+    }
+
+    return result;
+}
+
+UIntSetVal* UIntSetVal::fromUIntSet(ASTBuilder* astBuilder, const UIntSet& uintSet)
+{
+    return astBuilder->getUIntSetVal(uintSet);
+}
+
+void UIntSetVal::_toTextOverride(StringBuilder& out)
+{
+    out << "UIntSetVal{";
+    for (Index i = 0; i < getBitmaskCount(); i++)
+    {
+        if (i > 0)
+            out << ", ";
+        auto bitmask = getBitmask(i);
+        if (bitmask)
+        {
+            out << bitmask->getValue();
+        }
+        else
+        {
+            out << "null";
+        }
+    }
+    out << "}";
+}
+
+Val* UIntSetVal::_substituteImplOverride(ASTBuilder* astBuilder, SubstitutionSet subst, int* ioDiff)
```

Comment:
is this really necessary? would there ever be a case where the consituents of a UIntSetVal needs replacement? Aren't they always int literals?